File: AndroidStudioProjects/address/lib/address_book.dart
class AddressBook {
  final int? id;
  final String name;
  final String phone;
  final String email;
  final String address;

  AddressBook({
    this.id,
    required this.name,
    this.phone = '',
    this.email = '',
    this.address = '',
  });

  factory AddressBook.fromJson(Map<String, dynamic> j) => AddressBook(
    id: j['id'],
    name: j['name'] ?? '',
    phone: j['phone'] ?? '',
    email: j['email'] ?? '',
    address: j['address'] ?? '',
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'phone': phone,
    'email': email,
    'address': address,
  };
}

File: springwork/address_jpa/src/main/java/com/example/address/entity/AddressBook.java
package com.example.address.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "address_book")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AddressBook {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column(nullable = false)
	private String name;
	
	private String phone;
	private String email;
	private String address;
}
